plugins/scaffolder-backend-module-confluence-to-markdowwn/src/actions/confluence/confluenceToMarkdown.ts

```diff
@@ -0,0 +1,182 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+import { UrlReader } from '@backstage/backend-common';
+import { ScmIntegrations } from '@backstage/integration';
+import { createFetchPlainAction } from '@backstage/plugin-scaffolder-backend';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import { InputError, ConflictError } from '@backstage/errors';
+import { NodeHtmlMarkdown } from 'node-html-markdown';
+import fs from 'fs-extra';
+import parseGitUrl from 'git-url-parse';
+import {
+  parseFromYaml,
+  readFileAsString,
+  transformToYaml,
+  updateFile,
+  fetchConfluence,
+  getAndWriteAttachments,
+  createConfluenceVariables,
+} from './helpers';
+
+export const createConfluenceToMarkdownAction = (options: {
+  reader: UrlReader;
+  integrations: ScmIntegrations;
+  config: Config;
+}) => {
+  const { config, reader, integrations } = options;
+  const fetchPlainAction = createFetchPlainAction({ reader, integrations });
+  type Obj = {
+    [key: string]: string;
+  };
+
+  return createTemplateAction<{
+    confluenceUrls: string[];
+    repoUrl: string;
+  }>({
+    id: 'transform:confluence-to-markdown',
+    schema: {
+      input: {
+        properties: {
+          confluenceUrls: {
+            type: 'array',
+            title: 'Confluence URL',
+            description:
+              'Paste your confluence url. Ensure it follows this format: https://{confluence+base+url}/display/{spacekey}/{page+title}',
+            items: {
+              type: 'string',
+              default: 'Confluence URL',
+            },
+          },
+          repoUrl: {
+            type: 'string',
+            title: 'GitHub Repo Url',
+            description:
+              'mkdocs.yml file location inside the github repo you want to store the document',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const { confluenceUrls, repoUrl } = ctx.input;
+      const parsedRepoUrl = parseGitUrl(`${repoUrl}`);
+      const filePathToMkdocs = parsedRepoUrl.filepath.substring(
+        0,
+        parsedRepoUrl.filepath.lastIndexOf('/') + 1,
+      );
+      const dirPath = ctx.workspacePath;
+      let productArray: string[][] = [];
+
+      ctx.logger.info(`Fetching the mkdocs.yml catalog from ${repoUrl}`);
+
+      // This grabs the files from Github
+      const repoFileDir = `${dirPath}/${parsedRepoUrl.filepath}`;
+      await fetchPlainAction.handler({
+        ...ctx,
+        input: {
+          url: `https://${parsedRepoUrl.resource}/${parsedRepoUrl.owner}/${parsedRepoUrl.name}`,
+          targetPath: dirPath,
+        },
+      });
+
+      for (let p = 0; p < confluenceUrls.length; p++) {
```

Comment:
This can be converted to `for(const url of confluenceUrls) {`

Reply:
done